src/cpp/src/text_callback_streamer.cpp

```diff
@@ -46,7 +46,21 @@ bool TextCallbackStreamer::put(int64_t token) {
         res << std::string_view{text.data() + m_printed_len, print_until - m_printed_len} << std::flush;
         m_printed_len = print_until;
     }
-    return on_finalized_subword_callback(res.str());
+
+    return set_streaming_status(on_finalized_subword_callback(res.str()));
+}
+
+StreamingStatus TextCallbackStreamer::set_streaming_status(CallbackTypeVariant callback_status) {
+    StreamingStatus status = StreamingStatus::RUNNING;
+    if (auto res = std::get_if<StreamingStatus>(&callback_status)) {
+        status = *res;
+    } else if (auto res = std::get_if<bool>(&callback_status)) {
+        status = *res ? StreamingStatus::STOP : StreamingStatus::RUNNING;
+    } else if (auto res = std::get_if<std::monostate>(&callback_status)) {
+        status = StreamingStatus::RUNNING;
+    }
+
+    return status;
```

Comment:
```suggestion
    if (auto res = std::get_if<StreamingStatus>(&callback_status)) {
        return *res;
    if (auto res = std::get_if<bool>(&callback_status)) {
        return *res ? StreamingStatus::STOP : StreamingStatus::RUNNING;
    return StreamingStatus::RUNNING;
```
Same for utils and whisper bindings